src/cascadia/TerminalSettingsEditor/GlobalAppearance.cpp

```diff
@@ -2,19 +2,55 @@
 // Licensed under the MIT license.
 
 #include "pch.h"
+#include "Utils.h"
 #include "GlobalAppearance.h"
 #include "GlobalAppearance.g.cpp"
 #include "MainPage.h"
+#include "EnumEntry.h"
 
 using namespace winrt;
 using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
 using namespace winrt::Microsoft::Terminal::Settings::Model;
+using namespace winrt::Windows::Foundation::Collections;
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
-    GlobalAppearance::GlobalAppearance()
+    GlobalAppearance::GlobalAppearance() :
+        _ElementThemes{ winrt::single_threaded_observable_vector<Editor::EnumEntry>() }
     {
         InitializeComponent();
+
+        auto elementThemeMap = EnumMappings::ElementTheme();
+        for (auto [key, value] : elementThemeMap)
+        {
+            auto enumName = LocalizedNameForEnumName(L"Globals_Theme", key, L"Content");
+            auto entry = winrt::make<EnumEntry>(enumName, winrt::box_value<ElementTheme>(value));
+            _ElementThemes.Append(entry);
+
+            // Initialize the selected item to be our current setting
+            if (value == GlobalSettings().Theme())
+            {
+                ThemeButtons().SelectedItem(entry);
+            }
+        }

```

Comment:
(same comment about copy-pasta boilerplate here)

Reply:
_there must be a way to do this that makes it pure and stateless and doesn't require us to check `if value...` every time_